index.js

```diff
@@ -305,6 +306,22 @@ async function fastifyHttpProxy (fastify, opts) {
     wsProxy = setupWebSocketProxy(fastify, opts, rewritePrefix)
   }
 
+  function extractUrlComponents (urlString) {
+    const components = {
+      path: '',
+      queryParams: {}
+    }
+
+    const [path, queryString] = urlString.split('?')
```

Comment:
nit: you can put this in the beginning and initialize `components.path` directly with `path`


```js
const [path, queryString] = urlString.split('?')

   const components = {
      path,
      queryParams: {}
    }
```

Reply:
good call out should be updated